### ISSUE TITLE
Fetch or pull packages throw the error: remote already exists!

### ISSUE THREAD

**Issue by kguidonimartins** [radian-software/straight.el#978]
## What's wrong

Cannot update (fetch or pull) package updates. Even the `straight-freeze-versions` doesn't work. For example, running `(straight-fetch-package "ess")` return this message on `*straight-process*` buffer:

```
$ cd /home/karlo/.emacs.d.vanilla/straight/repos/ESS/
$ git remote add origin https\://github.com/emacs-ess/ESS.git

error: remote origin already exists.

[Return code: 3]
```

## Directions to reproduce

<details open><summary>Test Case</summary>

```emacs-lisp
(straight-bug-report
  :pre-bootstrap 
  (message "There is no problem in this step at all.")
  :post-bootstrap 
  (message "Installation is ok!")
  (straight-use-package
   '(ess))
  (message "But fetching/pull process always throw the error: remote origin already exists.")
  (straight-fetch-package "ess")
  (message "Maybe the problem is some git config on my side?"))
```
</details>

- Test run at: `2022-07-18 10:42:53`
- system-type: `gnu/linux`
- straight-version: `prerelease (HEAD -> master, origin/master, origin/HEAD) a90260ae 2022-07-18`
- emacs-version: `GNU Emacs 29.0.50 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.33, cairo version 1.17.4)
 of 2022-03-17`

<details><summary>Output</summary>

```emacs-lisp
There is no problem in this step at all.
Bootstrapping straight.el...
Bootstrapping straight.el...done
Looking for gnu-elpa-mirror recipe → Cloning melpa...
Looking for gnu-elpa-mirror recipe → Cloning melpa...done
Looking for nongnu-elpa recipe → Cloning gnu-elpa-mirror...
Looking for nongnu-elpa recipe → Cloning gnu-elpa-mirror...done
Looking for emacsmirror-mirror recipe → Cloning nongnu-elpa...
Looking for emacsmirror-mirror recipe → Cloning nongnu-elpa...done
Looking for emacsmirror-mirror recipe → Cloning el-get...
Looking for emacsmirror-mirror recipe → Cloning el-get...done
Looking for straight recipe → Cloning emacsmirror-mirror...
Looking for straight recipe → Cloning emacsmirror-mirror...done
Building straight...
Building straight...done

Test run with version: prerelease (HEAD -> develop, origin/develop) ba36462 2022-06-22
Installation is ok!
Cloning ESS (for ess)...
Cloning ESS (for ess)...done
Building ess...
Building ess...done

But fetching/pull process always throw the error: remote origin already exists.
Packages:
"straight"                n/a                  develop ba36462 2022-06-22
"org-elpa"                n/a                  n/a
"melpa"                   n/a                  master a6207f51 2022-07-17
"gnu-elpa-mirror"         n/a                  master aa9f053 2022-06-28
"nongnu-elpa"             n/a                  main e89ff131ae 2022-07-11
"el-get"                  melpa                master bf3dba44 2022-06-02
"emacsmirror-mirror"      n/a                  master ebaa7a9 2022-06-21
"ess"                     melpa                master a90260ae 2022-07-18

Error: (error "Failed to run \"git\"; see buffer *straight-process*")

```
</details>

### Version information

* Emacs version: `29.0.50`
* Operating system: `Arch Linux`

**Comment by kguidonimartins** [comment]
Sorry for bothering you with this, and thanks again for this amazing package. I just find that using `(straight-freeze-versions t)` solve my problem.

**Comment by raxod502** [comment]
Hum, that's an odd error you're getting, I do think that is a bug. However, when I tried, I got no error:


<details open><summary>Test Case</summary>

```emacs-lisp
(straight-bug-report
  :pre-bootstrap 
  (message "There is no problem in this step at all.")
  :post-bootstrap 
  (message "Installation is ok!")
  (straight-use-package
   '(ess))
  (message "But fetching/pull process always throw the error: remote origin already exists.")
  (straight-fetch-package "ess")
  (message "Maybe the problem is some git config on my side?"))
```
</details>

- Test run at: `2022-08-06 12:01:19`
- system-type: `gnu/linux`
- straight-version: `prerelease (HEAD -> develop, origin/develop) b6dba13 2022-06-12`
- emacs-version: `GNU Emacs 28.1 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.33, cairo version 1.16.0)
 of 2022-04-30`

<details><summary>Output</summary>

```emacs-lisp
There is no problem in this step at all.
Bootstrapping straight.el...
Bootstrapping straight.el...done
Looking for gnu-elpa-mirror recipe → Cloning melpa...
Looking for gnu-elpa-mirror recipe → Cloning melpa...done
Looking for nongnu-elpa recipe → Cloning gnu-elpa-mirror...
Looking for nongnu-elpa recipe → Cloning gnu-elpa-mirror...done
Looking for emacsmirror-mirror recipe → Cloning nongnu-elpa...
Looking for emacsmirror-mirror recipe → Cloning nongnu-elpa...done
Looking for emacsmirror-mirror recipe → Cloning el-get...
Looking for emacsmirror-mirror recipe → Cloning el-get...done
Looking for straight recipe → Cloning emacsmirror-mirror...
Looking for straight recipe → Cloning emacsmirror-mirror...done
Building straight...
Building straight...done

Test run with version: prerelease (HEAD -> develop, origin/develop) fed2153 2022-07-21
Installation is ok!
Cloning ESS (for ess)...
Cloning ESS (for ess)...done
Building ess...
Building ess...done

But fetching/pull process always throw the error: remote origin already exists.
Maybe the problem is some git config on my side?
Packages:
"straight"                n/a                  develop fed2153 2022-07-21
"org-elpa"                n/a                  n/a
"melpa"                   n/a                  master 1ed03e2c 2022-07-31
"gnu-elpa-mirror"         n/a                  master 55b898c 2022-07-26
"nongnu-elpa"             n/a                  main baad781689 2022-07-24
"el-get"                  melpa                master bf3dba44 2022-06-02
"emacsmirror-mirror"      n/a                  master 95df685 2022-07-30
"ess"                     melpa                master b8167727 2022-07-27


```
</details>

That suggests things might have been fixed somewhere between ba36462 (2022-06-22) and fed2153 (2022-07-21), but I think the only commit there was fed215348076ba9182f634e0770a175220474b5a, which shouldn't have changed anything.

Maybe there was a change in one of the upstream recipe repositories or packages that resolved the issue. One thing that might be nice is if `straight-bug-report` had a way to include/use a lockfile automatically to ensure reproducibility in a bug report.
